lib/github_changelog_generator/octo_fetcher.rb

```diff
@@ -345,7 +345,13 @@ def default_branch
     # @param [String] name
     # @return [Array<String>]
     def commits_in_branch(name)
-      @branches ||= client.branches(user_project).map { |branch| [branch[:name], branch] }.to_h
+      @branches ||= lambda do
+        iterate_pages(client, "branches") do |branches|
+          branches_map = branches.map { |branch| [branch[:name], branch] }.to_h
+          return branches_map if branches_map[name]
+        end
+        return {}
+      end[]
```

Comment:
Perhaps an explicit `#call` is clearer to more readers.

```suggestion
      end.call
```